library/spdm_requester_lib/libspdm_req_get_csr.c

```diff
@@ -93,16 +93,16 @@ static libspdm_return_t libspdm_try_get_csr(libspdm_context_t *spdm_context,
     spdm_request->opaque_data_length = opaque_data_length;
     spdm_request->requester_info_length = requester_info_length;
 
-    if (opaque_data_length != 0) {
+    if (requester_info_length != 0) {
         libspdm_copy_mem(spdm_request + 1,
                          spdm_request_size - sizeof(spdm_get_csr_request_t),
                          (uint8_t *)opaque_data, opaque_data_length);
```

Comment:
You did not change opaque_data here.

How that can pass the unit test?

Reply:
get_csr currently has no unit test related to this location, I will add them